src/main/java/dev/openfga/sdk/api/auth/AccessToken.java

```diff
@@ -15,32 +15,45 @@
 import static dev.openfga.sdk.util.StringUtil.isNullOrWhitespace;
 
 import java.time.Instant;
+import java.time.temporal.ChronoUnit;
 import java.util.Random;
 
 class AccessToken {
     private static final int TOKEN_EXPIRY_BUFFER_THRESHOLD_IN_SEC = 300;
-    private static final int TOKEN_EXPIRY_JITTER_IN_SEC =
-            300; // We add some jitter so that token refreshes are less likely to collide
-
-    private final Random random = new Random();
     private Instant expiresAt;
 
+    private final Random random = new Random();
     private String token;
 
     public boolean isValid() {
-        return !isNullOrWhitespace(token)
-                && (expiresAt == null
-                        || expiresAt.isAfter(Instant.now()
-                                .minusSeconds(TOKEN_EXPIRY_BUFFER_THRESHOLD_IN_SEC)
-                                .minusSeconds(random.nextLong() % TOKEN_EXPIRY_JITTER_IN_SEC)));
+        if (isNullOrWhitespace(token)) {
+            return false;
+        }
+
+        // Is expiry is null then the token will not expire so should be considered always valid
+        if (expiresAt == null) {
+            return true;
+        }
+
+        // A token should be considered valid until 5 minutes before the expiry with some jitter
+        // to account for multiple calls to `isValid` at the same time and prevent multiple refresh calls
+        Instant expiresWithLeeway = expiresAt
+                .minusSeconds(TOKEN_EXPIRY_BUFFER_THRESHOLD_IN_SEC)
+                .minusSeconds(random.nextInt(TOKEN_EXPIRY_BUFFER_THRESHOLD_IN_SEC))
```

Comment:
```suggestion
                .minusSeconds(random.nextInt(TOKEN_EXPIRY_JITTER_IN_SEC))
```
And re-add at the top
```
private static final int TOKEN_EXPIRY_JITTER_IN_SEC =
            300; // We add some jitter so that token refreshes are less likely to collide
```

to make it consistent w/ other SDKs if we change this field it will change appropriately

Reply:
Fixed in f98569c019551631f3f69b94e10d587f67fd6bc2